### PR TITLE
PPE change to sa_leaf in CanopyFluxesMod.F90

### LINKED ISSUE #2777: Missing sa_leaf code from PPE branch in CanopyFluxesMod.F90
### Brief summary of bug

The PPE branch (latest tag being PPE.n16_ctsm5.1.dev030) had a change to sa_leaf in CanopyFluxesMod.F90 when biomass heat storage is on.  This didn't make it into the latest CTSM tag.

### General bug information

**CTSM version you are using:** ctsm5.2.028

**Does this bug cause significantly incorrect results in the model's science?** Maybe.  The code mod was put in place to improve survivability, but our survivability in the current model in general is pretty good so it may not have that much effect.  Although maybe it will improve our C4 grass which was at 58% survivability in a recent simulation.

**Configurations affected:** All with biomass heat storage on.

### Details of bug

The relevant code block that was in the last PPE tag is:

```
            if(.not.(is_tree(patch%itype(p)) .or. is_shrub(patch%itype(p))) &
                 .or. dbh(p) < min_stem_diameter) then
               frac_rad_abs_by_stem(p) = 0.0
               sa_stem(p) = 0.0
!KO
               sa_leaf(p) = sa_leaf(p) + esai(p)
!KO
            endif

```

The line bracketed by !KO is not in the current code.
My notes from January 2021 detail a conversation with @dlawrenncar and @swensosc in which we were trying to improve survivability with biomass heat storage on.  After various tests we came up with this modification: "For grasses and skinny trees/shrubs, add in esai to sa_leaf".  I believe it was added for the situation where elai could be zero and thus sa_leaf would be zero (sa_leaf is set equal to 2*elai in code above this).

### PR DESCRIPTION
**Pull request by olyson**
### Description of changes

Add change to sa_leaf that was in PPE branch but is not on master

### Specific notes

Contributors other than yourself, if any:  @olyson 

CTSM Issues Fixed (include github issue #): 
Fixes #2777

Are answers expected to change (and if so in what way)?  Yes.  See #2777 for results.

Any User Interface Changes (namelist or namelist defaults changes)? No

Does this create a need to change or add documentation? Did you do so? Yes. No.

Testing performed, if any:
No test suite testing thus far.

### CODE REVIEW

**Review by slevis-lmwg — APPROVED** [review]
@olyson found this line missing from the recent PPE code that came to master. That recent code came in as b4b, while the modification here changes answers. Keith already evaluated the effect of this modification in #2777.

To me it seems a good idea to merge this quickly.

**Comment by samsrabin** [comment]
@wwieder Removing my assignment, as with the change to our queue order I no longer care when this comes in ;-)

**Comment by slevis-lmwg** [comment]
TODO slevis: Add [text from this post](https://github.com/ESCOMP/CTSM/pull/2820#discussion_r1792156681) to the ChangeLog.

**Comment by slevis-lmwg** [comment]
Starting aux_clm.

File: doc/ChangeLog
Comment (slevis-lmwg) [inline doc/ChangeLog:1]: @olyson I wanted to ask you for a less than 5-minute check of the ChangeLog.
It's probably fine. I'm just not confident marking all the configurations as "significant" answer changes. Still, that was my interpretation of your investigation of the changes in #2777.

Comment (slevis-lmwg) [reply doc/ChangeLog:1]: Thanks @olyson

**Comment by slevis-lmwg** [comment]
I have preemptively updated the ChangeLog with aux_clm results.
Izumi is done and came back OK.
Derecho is in progress and looks promising this far.

**Pull request merged**